src/bundles/pinning.js

```diff
@@ -103,35 +101,41 @@ const pinningBundle = {
     const remotePins = store.selectRemotePins().map(pin => pin.id)
     const notRemotePins = store.selectNotRemotePins()
 
-    // Check remaining CID status in chunks of 10 (remote API limitation)
-    const cids = uniqueCidBatches(allCids, 10)
+    // Check remaining CID status in chunks based on API limitation seen in real world
+    const cids = uniqueCidBatches(allCids, CID_PIN_CHECK_BATCH_SIZE)
 
     const adds = []
     const removals = []
 
     await Promise.allSettled(pinningServices.map(async service => {
       try {
         // skip CIDs that we know the state of at this service
-        const skipCids = new Set(
+        const skipCids = skipCache ? new Set() : new Set(
           [...remotePins, ...notRemotePins]
             .filter(id => id.startsWith(service.name))
             .map(cacheId2Cid)
         )
-        return Promise.allSettled(cids.map(async cidChunk => {
+        for (const cidChunk of cids) {
           const cidsToCheck = cidChunk.filter(cid => !skipCids.has(cid.toString()))
-          if (!cidsToCheck.length) return // skip if no new cids to check
+          if (!cidsToCheck.length) continue // skip if no new cids to check
           const notPins = new Set(cidsToCheck.map(cid => cid.toString()))
-          const pins = ipfs.pin.remote.ls({ service: service.name, cid: cidsToCheck })
-          for await (const pin of pins) {
-            const pinCid = pin.cid.toString()
-            notPins.delete(pinCid)
-            adds.push({ id: `${service.name}:${pinCid}`, ...pin })
+          try {
+            // TODO: execute remote.ls with +1m backoff per service when response Type == "error" and Message includes "429 Too Many Requests"
```

Comment:
:information_source: Curiously, hardcore caching is enough to send less than 6 requests per minute (I was not able to trigger rate-limiting at Pinata). 

If the user pins more than 6 things in one minute, they will get  a meaningful error via `notify.js` (yellow Toaster stating):

> Unable to set Pinning at Pinata: Error: reason: "Rate limit exceeded", details: "You have exceeded the API rate limit. Wait a minute and slow down your requests.": 429 Too Many Requests